components/ai_chat/core/common/mojom/ai_chat.mojom

```diff
@@ -147,6 +152,9 @@ struct ConversationTurn {
   // to be an event - as the events become richer, the order around text could
   // be important.
   array<ConversationEntryEvent>? events;
+
+  mojo_base.mojom.Time last_edited_time;
+  array<EditEntry> edits;
```

Comment:
Feels to me like this could be a `ConversationEntryEvent` instead of introducing a new property. Also potentially makes data storage simpler.

We could have a `UserSubmissionEvent` with a text, timestamp property.

Alternatively perhaps it's more appropriate to have an entirely new `ConversationTurn` for edits? That would allow:
- edited turns to have any properties / events that regular turns have
- simple data storage (in the upcoming SQL)
- potential for forking the conversation and maintaining history
This could happen via a property on ConversationTurn:
```
array<ConversationTurn> edits;
```

What do you think @yrliou?

Reply:
If we do the latter, we can definitely keep this PR to only allow a single conversation thread, but it would keep open the option to store and display multiple conversation forks, like competing products do.

Reply:
> I don't feel strongly that it has to be inside the events, having it outside seems intuitive to me rather than an entry in an array of unions. I wonder why would this make data storage harder, it doesn't seem very different than putting it in the events?

Because we are changing the structure from a simple `string text` property to an array of events. This has already been done for the assistant responses. And we already have storage setup for the events. Soon we'll either do the same for the user messages, or move away from ConversationTurn being the same struct for both assistant responses and user messages. If we're not going to do the below, I'm indifferent for the moment whether we keep this extra field of migrate to events. It doesn't have to be in this PR, we can do it in the storage PR.

> For later, currently it seems unnecessary to me to have it as a ConversationTurn, like, do we really need to edit other things in the turn?

Yes `ConversationTurn` is already a simple struct that has all the fields we might want to edit - including `action_type` and `text`. I can definitely see the user wanting to change action types when not getting the desired response. Seems better than duplicating fields in a separate struct for an Edit. And with `ConversationTurn` objects for edits, we can very simply construct the thread for each edit with a Parent / Child relationship if in the database each ConversationTurn maintains a `ParentConversationTurnId` field. This schema would prevent us having to modify the database schema when we want to preserve the pre-edit history and not delete it.

Can we try it here? I don't see that it will change anything too drastic?

If we want to support threading but keep the new field then we'll probably add a new `ConversationTurnEdits` table in the database, and maintain a relationship to `ConversationTurn` (aka `ConversationEntry`).

Reply:
I wasn't sure if we want to clone a complete turn struct, but I'm okay with it, I'm changing to the array<ConversationTurn> edits now.

Reply:
@petemill fixed in https://github.com/brave/brave-core/pull/24558/commits/a43b318311ca08abd78a7004f26c045608c24d81